orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/UpsertIntentTask.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.task
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.KeelService
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.keel.model.Intent
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentDryRunResponse
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentRequest
+import com.netflix.spinnaker.orca.keel.model.UpsertIntentResponse
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import java.util.HashMap
+import java.util.concurrent.TimeUnit
+
+@Component
+class UpsertIntentTask
+@Autowired constructor(
+  private val keelService: KeelService,
+  private val keelObjectMapper: ObjectMapper
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage) : TaskResult {
+    val missingParams = mutableListOf<String>()
+
+    if (!stage.context.containsKey("intents")) {
+      missingParams.add("intents")
+    }
+
+    if (!stage.context.containsKey("dryRun") ) {
+      missingParams.add("dryRun")
+    }
+
+    if (missingParams.isNotEmpty()) {
+      throw IllegalArgumentException("Missing required task parameters (${missingParams.joinToString(",")})")
+    }
+
+    val upsertIntentRequest = UpsertIntentRequest(
+      intents = keelObjectMapper.convertValue(stage.context["intents"], object : TypeReference<List<Intent>>() {}),
+      dryRun = if (stage.context["dryRun"] == null) false else stage.context["dryRun"].toString().toBoolean()
```

Comment:
do we really need to do `toString().toBoolean()` there? What is the value actually?

Reply:
The value _should_ be either `true` or `false`. `.toBoolean()` returns true if the value is equal to "true", ignoring case, and false otherwise. Since context is a map of `<String,Object>` I think we need to do `.toString()` first. 

I can simplify to `dryRun = stage.context["dryRun"].toString().toBoolean()` though.

Reply:
Is it a _string_ that's `"true"` or `"false"` or is it actually a boolean?

Knowing us it can probably be either!

Reply:
Should be an actual boolean (it's typed in keel). Gonna keep it as a boolean in the context to try and help with the untyped garbage.